### ISSUE TITLE
Migrate away from cfn-cmd

### ISSUE THREAD

**Issue by jonaf** [bazaarvoice/cloudformation-ruby-dsl#1]
Currently, we drop to the shell to evoke `cfn-cmd` for interaction with the Cloudformation API.

- It's cleaner, all within Ruby, and doesn't require separate configuration/installation of tools
- We can do more to refactor everything and be more object-oriented
- It fixes a few bugs (like having spaces in parameter or tag values)
- It's not deprecated :)

**Comment by jimcroft** [comment]
+1 

Would really like to see cfn-cmd replaced, ideally with use of the Ruby AWS SDK and support for the --profile flag to read from the same credentials file as the unified AWS CLI tools.

Happy to chip in here if this is an enhancement that would be supported.

**Comment by temujin9** [comment]
If you contribute a working PR, I will gladly help get it merged into
mainline. I dislike vendoring the underlying binaries, and direct API
access is much cleaner.

**Comment by jimcroft** [comment]
Cool. I have an open PR in the aws-sdk-ruby project (https://github.com/aws/aws-sdk-ruby/pull/703). I'll wait on an answer to that one before working on this as ideally I'd like the credential handling of cloudformation-ruby-dsl to match the AWS CLI (without re-inventing the wheel too much).

**Comment by temujin9** [comment]
Nice: looking forward to it!

**Comment by jdrago999** [comment]
Any word on this?

**Comment by amosshapira** [comment]
https://github.com/aws/aws-sdk-ruby/pull/703 was closed because the proposed code loads the region from the credentials file, which it shouldn't.

**Comment by jonaf** [comment]
Closed by #50
